### PR TITLE
Subobject classifier

### PR DESCRIPTION
**Pull request by mroman42**
Provides methods for the subobject classifier of a category.

1. In `UniversalObjects.gd`, declares the subobject classifier, the "true" morphism from the terminal object and the classifying morphism for a monomorphism.
2. In `UniversalObjects.gi`, implements the method `SubobjectClassifier` for a cell fo the category.
3. In `InstallAdds.gi`, creates functions similar to those of the terminal, initial and zero objects for the subobject classifier.
4. In `MethodRecord.gi`, adds records for these attributes.

The code will probably contain many obvious errors, as my knowledge of GAP is very limited. I will continue to improve this pull request, but a review would be really appreciated.

### CODE REVIEW

**Comment by mroman42** [comment]
There should be also a method for obtaining the monomorphism from the classifying morphism. It is a particular pullback, but maybe in particular cases a more efficient version can be implemented.

**Comment by mroman42** [comment]
@sebastianpos suggestions on this pull request

1. Replace `TruthMorphismIntoSubobjectClassifier` with `TruthMorphismIntoSubobjectClassifierWithGivenObjects(1,Omega)`. Then the former can be constructed as a function not in the method record. The implementation of `AssociatorLeftToRightWithGivenTensorProducts` is an example.
2. Declare both `ClassifyingMorphismOfSubobject` and `ClassifyingMorphismOfSubobjectWithGivenSubobjectClassifier`.
3. Derive `SubobjectOfClassifyingMorphism`.

File: CAP/gap/MethodRecord.gi [hunk 1]
```diff
@@ -621,6 +621,34 @@ UniversalMorphismFromInitialObjectWithGivenInitialObject := rec(
...
+ClassifyingMorphismOfSubobject := rec(
+  installation_name := "ClassifyingMorphismOfSubobject",
+  filter_list := [ "morphism" ],
+  return_type := "morphism" ),
+
+SubobjectOfClassifyingMorphism := rec(
+  installation_name := "SubobjectOfClassifyingMorphism",
+  filter_list := [ "morphism" ],
+  return_type := "morphism" ),
+
```
Comment (sebasguts) [inline CAP/gap/MethodRecord.gi]: I think it would be nice to add `io_type`s to the new methods.

File: CAP/gap/MethodRecord.gi [hunk 2]
```diff
@@ -621,6 +621,37 @@ UniversalMorphismFromInitialObjectWithGivenInitialObject := rec(
...
+ClassifyingMorphismOfSubobject := rec(
+  installation_name := "ClassifyingMorphismOfSubobject",
+  filter_list := [ "morphism" ],
+  io_type := [ [ "alpha" ] , [ "alpha_range" , "S" ] ],  
+  return_type := "morphism" ),
+
+SubobjectOfClassifyingMorphism := rec(
+  installation_name := "SubobjectOfClassifyingMorphism",
+  filter_list := [ "morphism" ],
+  io_type := [ [ "alpha" ] , [ "subobject" , "alpha_source" ] ]
```
Comment (zickgraf) [inline CAP/gap/MethodRecord.gi]: A comma is missing after this line.

**Pull request merged**